ui-tests/src/main/java/io/jenkins/plugins/analysis/warnings/IssuesTable.java

```diff
@@ -0,0 +1,168 @@
+     *
+     * @param link
+     *         the WebElement representing the link
+     * @param targetPageClass
+     *         the class of the PageObject representing the target page
+     *
+     * @return the PageObject representing the target page
+     */
+    public <T extends PageObject> T clickLinkOnSite(final WebElement link, final Class<T> targetPageClass) {
+        return resultDetailsPage.openLinkOnSite(link, targetPageClass);
+    }
+
+    public int getTotal() {
+        String tableInfo = issuesTab.findElement(By.id("issues_info")).getText();
+        String total = StringUtils.substringAfter(tableInfo, "of ");
+        return Integer.parseInt(StringUtils.substringBefore(total, " "));
+    }
+
+    /**
+     * Updates the table rows. E.g. if they are changed by toggling a details-row.
+     */
+    public void updateTableRows() {
+        tableRows.clear();
+        List<WebElement> tableRowsAsWebElements = tableElement.findElements(By.xpath(".//tbody/tr"));
+        tableRowsAsWebElements.forEach(element -> tableRows.add(getRightTableRow(element)));
+    }
+
+    /**
+     * Returns the table row as an object of the right sub class of {@link AbstractIssuesTableRow}.
+     *
+     * @param row
+     *         the WebElement representing the specific row.
+     *
+     * @return the table row
+     */
+    private AbstractIssuesTableRow getRightTableRow(final WebElement row) {
+        String rowType = row.getAttribute("role");
+        if (StringUtils.equals(rowType, "row")) {
+            if (type == IssuesTableRowType.DRY) {
+                return new DryIssuesTableRow(row, this);
+            }
+            else {
+                return new DefaultWarningsTableRow(row, this);
+            }
+        }
+        else {
+            return new DetailsTableRow(row);
+        }
+    }
+
+    /**
+     * Returns the amount of the headers for this table.
+     *
+     * @return the amount of table headers
+     */
+    public int getHeaderSize() {
+        return headers.size();
+    }
+
+    /**
+     * Returns the amount of table rows.
+     *
+     * @return the amount of table rows.
+     */
+    public int getSize() {
+        return tableRows.size();
+    }
+
+    /**
+     * Returns the table rows as List.
+     *
+     * @return the rows of the table
+     */
+    public List<AbstractIssuesTableRow> getTableRows() {
+        return tableRows;
+    }
+
+    /**
+     * Return the headers of the table.
+     *
+     * @return the headers
+     */
+    public List<String> getHeaders() {
+        return headers;
+    }
+
+    /**
+     * Returns a specific row as an instance of the expected class.
+     *
+     * @param row
+     *         the number of the row to be returned
+     * @param expectedClass
+     *         the expected type of the row
+     *
+     * @return the row
+     */
+    public <T extends AbstractIssuesTableRow> T getRowAs(final int row, final Class<T> expectedClass) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected @param tag for '<T>'.](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5070362)